Example-iOS-Swift/WebViewController.swift

```diff
@@ -78,16 +79,25 @@ class WebViewController: UIViewController, WKScriptMessageHandler {
     }
     
     func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
-        
         if let body = message.body as? String{
-            
             let jsonData = body.data(using: .utf8)
             
-            if let bodyStruct = try? JSONDecoder().decode(MessageData.self, from: jsonData!){
+            if let bodyStruct = try? JSONDecoder().decode(MessageData.self, from: jsonData!) {
                 switch bodyStruct.eventName {
                 case "v1.avatar.exported":
-                    avatarUrlDelegate?.avatarUrlCallback(url : bodyStruct.data!["url"]!)
+                    let event = AvatarExportedEvent (url: bodyStruct.data!["url"]!)
+                    webViewDelegate?.onAvatarExported(event: event)
                     reloadPage(clearHistory: false)
+                case "v1.asset.unlock":
+                    let event = AssetUnlockedEvent (userId: bodyStruct.data!["userId"]!,
+                                                    assetId: bodyStruct.data!["assetId"]!)
+                    webViewDelegate?.onAssetUnlocked(event: event)
+                case "v1.user.set":
```

Comment:
something I noticed yesterday working on the Kotlin example, there is also a v1.user.updated event. 
from what I could tell it seemed like the .updated event fires upon returning to avatar creator? 
Not sure exactly what condition triggers it though but in any case we should probably also handle that event.

Reply:
I just got the list from Sander, and it looks like there are others too. We should add them to both 👍 -> https://docs.readyplayer.me/ready-player-me/integration-guides/web-and-native-integration/avatar-creator-integration#emitted-events